lib/FetchEnvs.js

```diff
@@ -288,7 +288,14 @@ function typeCast(value, type) {
       return (value.toLowerCase() === 'true');
     }
     case 'json': {
-      return JSON.parse(value);
+      if (value) {
+        try {
+          return JSON.parse(value);
+        } catch (error) {
+          throw new Error(`JSON invalid in ref ${name}: ${value}. Parse error: (${error})`)
+        }
+      }
+      return {}
```

Comment:
im not sure im on board with this being the default return value here. Just because a user defined type as json, why does that translate a user defined default value of `'""` into `{}`? If a user wants the default value to be `{}`, why not make them specify it explicitly?

Reply:
On line 291 it returns undefined if value is null, but will return {} for any other falsey value. We should consider how to handle these scenarios for the diff types

Reply:
oohhh.. i understand now.. i was thinking about where this was in the process wrong. this will allow users to not break there mtp when they go to lookup a value in the template (allows `my.value` to be undefined instead of an error)

Reply:
```suggestion
      return {};
```

Reply:
fixed in [13520bb](https://github.com/razee-io/razeedeploy-core/pull/311/commits/13520bbd05d974f3d1d399b490ea6f45a9f3b32f)